src/Draco.Compiler/Api/Syntax/SyntaxFactory.cs

```diff
@@ -103,11 +110,27 @@ public static SeparatedSyntaxList<TNode> SeparatedSyntaxList<TNode>(SyntaxToken
             (path, member) => MemberImportPath(path, Dot, Name(member))),
         Semicolon);
 
+    public static FunctionDeclarationSyntax FunctionDeclaration(
+        SyntaxToken? visibility,
```

Comment:
I'd still suggest changing this overload to use Visibility instead of a token, as this is a higher level API (using strings instead of tokens, ...) and you can just use VisibilityToken for impl